python/google/protobuf/pyext/descriptor.cc

```diff
@@ -132,7 +169,7 @@ bool _CalledFromGeneratedFile(int stacklevel) {
     return false;
   }
 
-  if (frame->f_globals != frame->f_locals) {
+  if (PyFrame_GetGlobals(frame) != PyFrame_GetLocals(frame)) {
```

Comment:
this appears to be purely testing for at global scope.  How about the helper does that directly

Reply:
I believe this falls in the same category, the helpers are for backwards compatibility so we need to keep them consistent with the new API in Python 3.11